File: pkgts/settingstore/consts.ts
import {
	anthropicProviderInfo,
	googleProviderInfo,
	huggingfaceProviderInfo,
	llamacppProviderInfo,
	openaiProviderInfo,
	ProviderName,
} from 'aiprovider';
import { SettingsSchema } from './types';

export const defaultAISettings = {
	[ProviderName.ANTHROPIC]: {
		isEnabled: true,
		apiKey: anthropicProviderInfo.apiKey,
		defaultModel: anthropicProviderInfo.defaultModel,
		defaultOrigin: anthropicProviderInfo.defaultOrigin,
		defaultTemperature: anthropicProviderInfo.defaultTemperature,
		additionalSettings: anthropicProviderInfo.additionalSettings,
	},
	[ProviderName.GOOGLE]: {
		isEnabled: false,
		apiKey: googleProviderInfo.apiKey,
		defaultModel: googleProviderInfo.defaultModel,
		defaultOrigin: googleProviderInfo.defaultOrigin,
		defaultTemperature: googleProviderInfo.defaultTemperature,
		additionalSettings: googleProviderInfo.additionalSettings,
	},
	[ProviderName.HUGGINGFACE]: {
		isEnabled: false,
		apiKey: huggingfaceProviderInfo.apiKey,
		defaultModel: huggingfaceProviderInfo.defaultModel,
		defaultOrigin: huggingfaceProviderInfo.defaultOrigin,
		defaultTemperature: huggingfaceProviderInfo.defaultTemperature,
		additionalSettings: huggingfaceProviderInfo.additionalSettings,
	},
	[ProviderName.LLAMACPP]: {
		isEnabled: false,
		apiKey: llamacppProviderInfo.apiKey,
		defaultModel: llamacppProviderInfo.defaultModel,
		defaultOrigin: llamacppProviderInfo.defaultOrigin,
		defaultTemperature: llamacppProviderInfo.defaultTemperature,
		additionalSettings: llamacppProviderInfo.additionalSettings,
	},
	[ProviderName.OPENAI]: {
		isEnabled: true,
		apiKey: openaiProviderInfo.apiKey,
		defaultModel: openaiProviderInfo.defaultModel,
		defaultOrigin: openaiProviderInfo.defaultOrigin,
		defaultTemperature: openaiProviderInfo.defaultTemperature,
		additionalSettings: openaiProviderInfo.additionalSettings,
	},
};

export const defaultSettingsData: SettingsSchema = {
	app: {
		defaultProvider: ProviderName.OPENAI,
	},
	aiSettings: defaultAISettings,
};

export const sensitiveKeys = [
	'aiSettings.openai.apiKey',
	'aiSettings.anthropic.apiKey',
	'aiSettings.huggingface.apiKey',
	'aiSettings.google.apiKey',
	'aiSettings.llamacpp.apiKey',
];

File: pkggo/settingstore/spec/consts.go
package spec

import (
	"github.com/flexigpt/flexiui/pkggo/aiprovider/anthropic"
	"github.com/flexigpt/flexiui/pkggo/aiprovider/google"
	"github.com/flexigpt/flexiui/pkggo/aiprovider/huggingface"
	"github.com/flexigpt/flexiui/pkggo/aiprovider/llamacpp"
	"github.com/flexigpt/flexiui/pkggo/aiprovider/openai"
	aiproviderSpec "github.com/flexigpt/flexiui/pkggo/aiprovider/spec"
)

// Define the default AI settings.
var DefaultAISettings = map[aiproviderSpec.ProviderName]AISetting{
	anthropic.ProviderNameAnthropic: {
		IsEnabled:          true,
		ApiKey:             anthropic.AnthropicProviderInfo.ApiKey,
		DefaultModel:       anthropic.AnthropicProviderInfo.DefaultModel,
		DefaultOrigin:      anthropic.AnthropicProviderInfo.DefaultOrigin,
		DefaultTemperature: anthropic.AnthropicProviderInfo.DefaultTemperature,
		AdditionalSettings: anthropic.AnthropicProviderInfo.AdditionalSettings,
	},
	google.ProviderNameGoogle: {
		IsEnabled:          false,
		ApiKey:             google.GoogleProviderInfo.ApiKey,
		DefaultModel:       google.GoogleProviderInfo.DefaultModel,
		DefaultOrigin:      google.GoogleProviderInfo.DefaultOrigin,
		DefaultTemperature: google.GoogleProviderInfo.DefaultTemperature,
		AdditionalSettings: google.GoogleProviderInfo.AdditionalSettings,
	},
	huggingface.ProviderNameHuggingFace: {
		IsEnabled:          false,
		ApiKey:             huggingface.HuggingfaceProviderInfo.ApiKey,
		DefaultModel:       huggingface.HuggingfaceProviderInfo.DefaultModel,
		DefaultOrigin:      huggingface.HuggingfaceProviderInfo.DefaultOrigin,
		DefaultTemperature: huggingface.HuggingfaceProviderInfo.DefaultTemperature,
		AdditionalSettings: huggingface.HuggingfaceProviderInfo.AdditionalSettings,
	},
	llamacpp.ProviderNameLlamaCPP: {
		IsEnabled:          false,
		ApiKey:             llamacpp.LlamacppProviderInfo.ApiKey,
		DefaultModel:       llamacpp.LlamacppProviderInfo.DefaultModel,
		DefaultOrigin:      llamacpp.LlamacppProviderInfo.DefaultOrigin,
		DefaultTemperature: llamacpp.LlamacppProviderInfo.DefaultTemperature,
		AdditionalSettings: llamacpp.LlamacppProviderInfo.AdditionalSettings,
	},
	openai.ProviderNameOpenAI: {
		IsEnabled:          true,
		ApiKey:             openai.OpenAIProviderInfo.ApiKey,
		DefaultModel:       openai.OpenAIProviderInfo.DefaultModel,
		DefaultOrigin:      openai.OpenAIProviderInfo.DefaultOrigin,
		DefaultTemperature: openai.OpenAIProviderInfo.DefaultTemperature,
		AdditionalSettings: openai.OpenAIProviderInfo.AdditionalSettings,
	},
}

// Define the default settings data.
var DefaultSettingsData = SettingsSchema{
	App: AppSettings{
		DefaultProvider: openai.ProviderNameOpenAI,
	},
	AISettings: DefaultAISettings,
}

// Define the sensitive keys.
var SensitiveKeys = []string{
	"aiSettings.openai.apiKey",
	"aiSettings.anthropic.apiKey",
	"aiSettings.huggingface.apiKey",
	"aiSettings.google.apiKey",
	"aiSettings.llamacpp.apiKey",
}
